interface/admin.py

```diff
@@ -1,6 +1,48 @@
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.db.models import F
+from django.http import HttpResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        assignment = _assignment[0]
+
+        subs = Submission.objects.filter(
+                assignment=assignment).order_by(F('timestamp').asc())
+
+        submissions = {}
+
+        for submission in subs:
+            submissions[submission.user.username] = submission
+
+        with TemporaryDirectory() as _tmp:
+            tmp = Path(_tmp)
+
+            with ZipFile(tmp / 'review.zip', 'x') as zipfile:
+                for _, submission in submissions.items():
+                    submission.download(tmp / f'{submission.id}.zip')
+                    zipfile.write(
+                        tmp / f'{submission.id}.zip',
+                        f'{submission.user.username}-{submission.id}.zip')
+
+            with open(tmp / 'review.zip', 'rb') as zipfile:
+                response = HttpResponse(zipfile.read(),
+                                        content_type='application/zip')
```

Comment:
https://docs.djangoproject.com/en/2.2/ref/request-response/#fileresponse-objects

It expects an open file from which it will stream the content. Your zip file will be unlinked when you exit the `with TemporaryDirectory` block, but thats' ok, you can open a filehandle on `tmp / 'review.zip'`, pass it on to `FileResponse`, and the actual file on disk will be removed once that filehandle is closed.

```python
with TemporaryDirectory() as _tmp:
    # ...
    review_zip = (tmp / 'review.zip').open('rb')
# TemporaryDirectory deletes itself and the `review.zip` hard link,
# but the kernel knows, you have the file open, so it doesn't
# delete the inode quite yet.
return FileResponse(review_zip)
```